### PR TITLE
Fix password renew for users without core.admin permission

### PR DESCRIPTION
**Pull request by chmst**
Pull Request for Issue #39028 .

### Summary of Changes
Set a redirect to the backend homepage if a user has to renew a password but has not core.admin permission

### Testing Instructions
see #39028 .

### Actual result BEFORE applying this Pull Request
see #39028

### Expected result AFTER applying this Pull Request
The user is redirected to the dashboard.

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/components/com_users/src/Controller/UserController.php [hunk 1]
```diff
@@ -117,6 +117,12 @@ public function save($key = null, $urlVar = null)
             $this->setRedirect($return);
         }
 
+        // If a user has to renew a password but has no permission for users
+        if (!$this->app->getIdentity()->authorise('core.admin', 'com_users'))
+        {
```
Comment (richard67) [inline administrator/components/com_users/src/Controller/UserController.php]:
```suggestion
        if (!$this->app->getIdentity()->authorise('core.admin', 'com_users')) {
```
Fix PHPCS failing in drone.

**Comment by MacJoom** [comment]
I have tested this item :white_check_mark: successfully on 8284fed57264ddf537a15e40915f6ea481f652b8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39038">issues.joomla.org/tracker/joomla-cms/39038</a>.</sub>

**Comment by MacJoom** [comment]
I have tested this item :white_check_mark: successfully on 8284fed57264ddf537a15e40915f6ea481f652b8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39038">issues.joomla.org/tracker/joomla-cms/39038</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 8284fed57264ddf537a15e40915f6ea481f652b8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39038">issues.joomla.org/tracker/joomla-cms/39038</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39038">issues.joomla.org/tracker/joomla-cms/39038</a>.</sub>

File: administrator/components/com_users/src/Controller/UserController.php [hunk 2]
```diff
@@ -117,6 +117,11 @@ public function save($key = null, $urlVar = null)
             $this->setRedirect($return);
         }
 
+        // If a user has to renew a password but has no permission for users
+        if (!$this->app->getIdentity()->authorise('core.admin', 'com_users')) {
```
Comment (laoneo) [inline administrator/components/com_users/src/Controller/UserController.php:121]: Should this not only be executed when $result is false?

Comment (chmst) [reply administrator/components/com_users/src/Controller/UserController.php:121]: I understand that the result of the password reset must be true. But after password change the user cannot remain on com_users

Comment (roland-d) [reply administrator/components/com_users/src/Controller/UserController.php:121]: @laoneo The value of $result is irrelevant I think, it is about whether the user has access to the `com_users`. Why do you think the value is relevant?

**Comment by fancyFranci** [comment]
Thank you very much :)

**Pull request merged**